source/unified-test-format/unified-test-format.rst

```diff
@@ -582,6 +583,58 @@ The structure of this object is as follows:
 
   - ``serverApi``: Optional `serverApi`_ object.
 
+.. _entity_encryptedClient:
+
+- ``encryptedClient``: Optional object. Defines a ClientEncryption object.
+
+  The structure of this object is as follows:
+
+  - ``id``: Required string. Unique name for this entity. The YAML file SHOULD
+    define a `node anchor`_ for this field (e.g.
+    ``id: &encryptedClient0 encryptedClient0``).
+
+  - ``clientEncryptionOpts``: Required document. A value corresponding to a
+    `ClientEncryptionOpts
+    <../client-side-encryption/client-side-encryption.rst#clientencryption>`__.
+
+    The structure of this document is as follows:
+
+    - ``keyVaultClient``: Required string. MongoClient entity from which this
```

Comment:
> I now see that it states [under keyVaultClient](https://github.com/mongodb/specifications/blob/0142955943d91a783fc0314fb41a14e392bd15b6/source/client-side-encryption/client-side-encryption.rst#keyvaultclient) that the field is indeed optional

`keyVaultClient` is optional for `AutoEncryptionOpts`. It is required for `ClientEncryptionOpts`.

`ClientEncryption` requires a `keyVaultClient` because it does not configure its own internal MongoClient. The `keyVaultClient` is the only way it can access a MongoDB server.

Reply:
I want to confirm: were @jmikola and I confusing `AutoEncryptionOpts.keyVaultClient` with `ClientEncryptionOpts.keyVaultClient` when stating that `keyVaultClient` is an optional field? If so, would it be preferable to specify the (required) key vault client via `ClientEncryptionOpts.keyVaultClient` as was being done prior to https://github.com/mongodb/specifications/commit/5f8b40343751444de70fb77a7104c0b77cdc006a rather than a `client` property?

Reply:
> I want to confirm: were @jmikola and I confusing AutoEncryptionOpts.keyVaultClient with ClientEncryptionOpts.keyVaultClient when stating that keyVaultClient is an optional field? 

Yes.

> If so, would it be preferable to specify the (required) key vault client via ClientEncryptionOpts.keyVaultClient as was being done prior to https://github.com/mongodb/specifications/commit/5f8b40343751444de70fb77a7104c0b77cdc006a rather than a client property?

Either seems OK. I have a slight preference for using `ClientEncryptionOpts.keyVaultClient` for consistency with the CSFLE spec API.

Reply:
@jmikola Would like to confirm we are on the same page before reverting related changes.

Reply:
Ah, I might have been confused myself as in PHPC we construct ClientEncryption objects from our MongoClient equivalent (i.e. [`MongoDB\Driver\Manager::createClientEncryption()`](https://www.php.net/manual/en/mongodb-driver-manager.createclientencryption.php)). For that reason, we document `keyVaultClient` as option since there is always a client context available to fall back on.

I wasn't aware that other drivers allow direct construction of ClientEncryption objects. I don't intend to change PHPC's behavior here, but would it make sense to revise the spec to permit it being optional if/when a ClientEncryption is created via a MongoClient?

Reply:
Regardless of my previous point, I think the field _within_ `clientEncryptionOpts` should be named `keyVaultClient` for consistency with the CSFLE spec. The name `client` makes sense for any sibling field of `clientEncryptionOpts`, but per @kevinAlbs' comment above it sounds like there's no need for that.

Reply:
Agreed. Reverted from `client` field under `clientEncryption` to the previous required `keyVaultClient` field under `clientEncryptionOpts` for consistency with CSE spec.

> So perhaps we can add a comment to the `keyVaultClient` field within `ClientEncryptionOpts` to note that drivers MAY make it optional if ClientEncryption is constructed from a MongoClient object.

Prior confusion aside, I think the CSE spec is clear that a ClientEncryption object is constructed with a `ClientEncryptionOpts` argument that specifies the MongoClient to be constructed with via (a required) `keyVaultClient` in the same manner that a MongoClient is constructed with an `AutoEncryptionOpts` argument that (optionally) specifies a separate `keyVaultClient`. It sounds like PHPC architecturally designed ClientEncryption as spawning from a MongoClient (so naturally, said MongoClient is assumed to be the `keyVaultClient` by default), but I don't think such a coupling is required by the CSE spec. I think it would be more appropriate to make an effort to decouple the ClientEncryption object from the MongoClient object in PHPC than to adjust the CSE spec to strengthen the coupling. But either way, this falls outside the scope of this PR, so as you suggested, if you want to pursue an edit to the CSE spec, we can continue the discussion in a separate DRIVERS ticket.

Reply:
Created [DRIVERS-2296](https://jira.mongodb.org/browse/DRIVERS-2296) to track that.